lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -371,11 +371,11 @@ public function setParameters($parameters)
     /**
      * Sets a query parameter.
      *
-     * @param string|int  $key   The parameter position or name.
-     * @param mixed       $value The parameter value.
-     * @param string|null $type  The parameter type. If specified, the given value will be run through
-     *                           the type conversion of this type. This is usually not needed for
-     *                           strings and numeric types.
+     * @param string|int      $key   The parameter position or name.
+     * @param mixed           $value The parameter value.
+     * @param string|int|null $type  The parameter type. If specified, the given value will be run through
```

Comment:
Should we make this more precise?
```suggestion
     * @param string|Connection::PARAM_*|ParameterType::*|null $type  The parameter type. If specified, the given value will be run through
```

Reply:
Is that something IDE's and static analyzers understand? :thinking:

Reply:
Static analyzers, yes. IDE, I don't know (I use neovim). Please test if you can.

Reply:
AFAIK, we use this kind of literal type syntax only in annotations with `psalm-` vendor prefix, e.g. `@psalm-param` in this case.

Reply:
Anyway, let's not go down that road in a bugfix PR.